### ISSUE TITLE
Snowflake - Fix Error when semicolon is preceeded by space

### ISSUE THREAD

**Issue by WittierDinosaur** [sqlfluff/sqlfluff#2827]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff identifies fixable error - and fails to fix it

### Expected Behaviour

Semicolon snaps to end of statement

### Observed Behaviour

No semicolon identified at end of statement, unable to fix

### How to reproduce

sqlfluff fix test.sql --dialect=snowflake

```
SET foo = (SELECT foo FROM foo.foo) ;

```

### Dialect

Snowflake

### Version

Current HEAD of master

### Configuration

Default config - but snowflake

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
I do see this in the verbose

```
INFO       Applying Fixes [L052]: [<LintFix: delete @[L:  1, P: 13] delete:';'>, <LintFix: create_after @[L:  1, P: 12] create:';'>, <LintFix: delete @[L:  1, P: 12] delete:' '>]
```

That is, it is creating the fix after the statement (anchored at the **end** of the statement), then deleting the space and the old semi-colon.

Whereas for a simple `SELECT` statement (that does work) it has this:

```
INFO       Applying Fixes [L052]: [<LintFix: create_after @[L:  1, P:  1] create:';'>, <LintFix: delete @[L:  1, P: 22] delete:';'>, <LintFix: delete @[L:  1, P: 21] delete:' '>] 
```

That is, it is creating the fix after the statement (anchored at the **beginning** of the statement), then deleting the space and the old semi-colon.

So seems to be a problem with `_choose_anchor_segment` for `SET` statements...

**Comment by tunetheweb** [comment]
So it works if you add an `Indent`/`Dendent` to the `SET` statement:

```py
    match_grammar = OneOf(
        Sequence(
            "SET",
            Indent,
            Ref("LocalVariableNameSegment"),
            Ref("EqualsSegment"),
            Ref("ExpressionSegment"),
            Dedent,
        ),
```

Guessing `_choose_anchor_segment` does something funky there.

However also discovered `create_before` works fine and avoids the issue! So will do a PR for that.

**Comment by tunetheweb** [comment]
> However also discovered `create_before` works fine and avoids the issue! So will do a PR for that.

Actually that’s not a complete fix as it leaves an extra whitespace after the semicolon. That is fixed by a subsequent pass by another rules but not ideal.

@barrywhart would be good to get your opinion here as so think there’s something funny about how the meta is handled compared to statements with no meta that maybe isn’t quite the right thing to do. Perhaps related to your recent change here?

**Comment by barrywhart** [comment]
Quick update: It had the same issue before yesterday's fix.

**Comment by barrywhart** [comment]
It also fails if I eliminate the call to `_choose_anchor_segment()`. This rule is proving to be the new L003...

**Comment by barrywhart** [comment]
I think this is the issue. The core linter can't handle multiple fixes from the same rule with the same anchor, yet L052 is doing that.
```
(Pdb) pp fixes
[<LintFix: create_after @[L:  1, P: 36] create:';'>,
 <LintFix: delete @[L:  1, P: 37] delete:';'>,
 <LintFix: delete @[L:  1, P: 36] delete:' '>]
(Pdb) pp [fix.anchor for fix in fixes]
[<WhitespaceSegment: ([L:  1, P: 36]) ' '>,
 <SymbolSegment: ([L:  1, P: 37]) ';'>,
 <WhitespaceSegment: ([L:  1, P: 36]) ' '>]
```

I'll try and fix the L052 and maybe update the core linter to detect and warn or fail on this. Currently, it just silently drops one of the fixes, which is not good.

**Comment by barrywhart** [comment]
I have a PR which fixes this. I added the core linter check I mentioned (duplicate anchors), and it's catching a similar bug in L036. I'll try and fix that in the same PR, as long as it's not a big nasty fix.
